monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -0,0 +1,389 @@
+            logger.info(repr(normal_out).replace("\\n", "\n").replace("\\t", "\t"))
+        except subprocess.CalledProcessError as e:
+            output = repr(e.stdout).replace("\\n", "\n").replace("\\t", "\t")
+            errors = repr(e.stderr).replace("\\n", "\n").replace("\\t", "\t")
+            raise RuntimeError(f"subprocess call error {e.returncode}: {errors}, {output}") from e
+        return normal_out
+
+    def get_score(self, *args, **kwargs):
+        """
+        Returns validation scores of the model trained by the current Algo.
+        """
+        config_yaml = os.path.join(self.output_path, "configs", "algo_config.yaml")
+        parser = ConfigParser()
+        parser.read_config(config_yaml)
+        ckpt_path = parser.get_parsed_content("ckpt_path", default=self.output_path)
+
+        dict_file = ConfigParser.load_config_file(os.path.join(ckpt_path, "progress.yaml"))
+        return dict_file["best_avg_dice_score"]  # TODO: define the format of best scores' list and progress.yaml
+
+    def get_inferer(self, *args, **kwargs):
+        """
+        Load the InferClass from the infer.py
+        """
+        infer_py = os.path.join(self.output_path, "scripts", "infer.py")
+        if not os.path.isfile(infer_py):
+            raise ValueError(f"{infer_py} is not found, please check the path.")
+        config_path = os.path.join(self.output_path, "configs", "algo_config.yaml")
+        config_path = config_path if os.path.isfile(config_path) else None
+        logger.info(f"in memory or persistent predictions using {config_path}.")
+        spec = importlib.util.spec_from_file_location("InferClass", infer_py)
+        infer_class = importlib.util.module_from_spec(spec)
+        sys.modules["InferClass"] = infer_class
+        spec.loader.exec_module(infer_class)
+        return infer_class.InferClass(config_path, *args, **kwargs)
+
+    def predict(self, predict_params=None):
+        """
+        Use the trained model to predict the outputs with a given input image. Path to input image is in the params
+        dict in a form of {"files", ["path_to_image_1", "path_to_image_2"]}. If it is not specified, then the pre-
+        diction will use the test images predefined in the bundle config.
+
+        Args:
+            predict_params: a dict to override the parameters in the bundle config (including the files to predict).
+
+        """
+        if predict_params is None:
+            params = {}
+        else:
+            params = deepcopy(predict_params)
+
+        files = params.pop("files", ".")
+        inferer = self.get_inferer(**params)
+        return [inferer.infer(f) for f in ensure_tuple(files)]
+
+
+# path to download the algo_templates
+algo_zip = "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/algo_templates.tar.gz"
+
+md5 = "67f434ce3c44800cfb6341d6278d635b"
+
+# default algorithms
+default_algos = {
+    "unet": dict(
```

Comment:
is this unet the same as the one in gitlab repo?

Reply:
I did not validate but tried my best to replicate the all the details.